mediatek: mt7622: fix banana pi r64 wps button

PolynomialDivision opened the pull request:
Fix the wps button to prevent wrongly detected recovery procedures.
In the official banana pi r64 git the wps button is set to
GPIO_ACTIVE_LOW and not GPIO_ACTIVE_HIGH.

Fixes on boot unwanted recovery entering:

  Press the [f] key and hit [enter] to enter failsafe mode
  Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
  - failsafe button wps was pressed -
  - failsafe -

Refresh patches:
- 402-mtd-blktrans-call-add-disks-after-mtd-device.patch
- 420-mtd-set-rootfs-to-be-root-dev.patch
- 495-mtd-core-add-get_mtd_device_by_node.patch
- 510-net-mediatek-add-flow-offload-for-mt7623.patch

musashino205 commented:
Isn't it caused by the misconfigured gpio property? Did you try changing ```GPIO_ACTIVE_HIGH``` to ```GPIO_ACTIVE_LOW```?

I checked the [official repository for R64](https://github.com/BPI-SINOVOIP/BPI-R64-openwrt/blob/f57c12d6e6c6002af19322f0baf55a7eac770c94/target/linux/mediatek/files-4.14/arch/arm64/boot/dts/mediatek/mt7622-bananapi-bpi-r64.dts#L52), it was defiend as ```GPIO_ACTIVE_LOW```.

PolynomialDivision commented:
It seems to fix it. At least I tried to provoke the issue again and I was not sucesfull. WPS recovery still works when pressing the button. I would send it upstream and exchange the patch with upstream patch? @musashino205 I would put you as `Suggested-By: Signed-off-by: INAGAKI Hiroshi <musashino.open@gmail.com>`?

musashino205 commented:
Yeah, great!

> I would send it upstream and exchange the patch with upstream patch? @musashino205 I would put you as Suggested-By: INAGAKI Hiroshi <musashino.open@gmail.com>?

Sounds great, and it's OK for me :)

PolynomialDivision commented:
> > Why did they change that on upstream Linux? Thanks! I will immediately test it.
> 
> I'm curious can you try to find the commit that made the change upstream?

I would also be very interested! :) However, the first commit on linux master directly introduced it "wrong":
https://lore.kernel.org/lkml/69a61903d53e5a818d1ddabd0613d237217c62a8.1534607107.git.ryder.lee@mediatek.com/

I exchanged now the patch with the one I sent upstream. Can we merge or should we wait what upstream linux says?
http://lists.infradead.org/pipermail/linux-mediatek/2022-June/044533.html

Ansuel commented:
> I would also be very interested! :) However, the first commit on linux master directly introduced it "wrong":

Then it was probably a typo.

> I exchanged now the patch with the one I sent upstream. Can we merge or should we wait what upstream linux says?

I would wait upstream but for this simple change i think we can merge this... Anyway can you rebase this and split the changes?
In one commit you make the gpio change and also refresh patches. Pls split them or drop the patch refresh.

PolynomialDivision committed:
mediatek: mt7622: fix banana pi r64 wps button

Fix the wps button to prevent wrongly detected recovery procedures.
In the official banana pi r64 git the wps button is set to
GPIO_ACTIVE_LOW and not GPIO_ACTIVE_HIGH.

Import patch to fix on boot unwanted recovery entering:

  Press the [f] key and hit [enter] to enter failsafe mode
  Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
  - failsafe button wps was pressed -
  - failsafe -

Signed-off-by: Nick Hainke <vincent@systemli.org>

Ansuel commented:
@PolynomialDivision I know you will hate me

split the commit to 
```
generic: 5.15: refresh patches
mediatek: 5.15: refresh patches 
```

PolynomialDivision committed:
generic: 5.15: refresh patches

Refresh patches:
- 402-mtd-blktrans-call-add-disks-after-mtd-device.patch
- 420-mtd-set-rootfs-to-be-root-dev.patch
- 495-mtd-core-add-get_mtd_device_by_node.patch

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
mediatek: 5.15: refresh patches

Refresh patches:
- 510-net-mediatek-add-flow-offload-for-mt7623.patch
- 920-dts-mt7622-bpi-r64-fix-wps-button.patch

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
> @PolynomialDivision I know you will hate me

No. I really appreciate it that you are reviewing my patch and helping me. :) Updated.

Ansuel commented:
Thanks, merged in master with:
04545c4325 mediatek: 5.15: refresh patches
3c3367f03e generic: 5.15: refresh patches
6686194255 mediatek: mt7622: fix banana pi r64 wps button

PolynomialDivision commented:
@Ansuel Nice thanks. I will do a PR when it arrived really in the next tree. :)
Still missing here: https://git.kernel.org/pub/scm/linux/kernel/git/matthias.bgg/linux.git/log/?h=for-next
